x-pack/plugins/security_solution/server/utils/runtime_types.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { either, fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+import * as rt from 'io-ts';
+import { failure } from 'io-ts/lib/PathReporter';
+import get from 'lodash/get';
+
+type ErrorFactory = (message: string) => Error;
+
+export type GenericIntersectionC =
```

Comment:
I know but I was not able to win this fight with typescript if you have a better idea, I am here to hear it.